src/_utils/datetime-utils.js

```diff
@@ -0,0 +1,149 @@
+const digitRegex = /^[0-9]$/;
+const formatSpecifierRegex = /%[HMdmyY%]/g;
+const daysInWeek = 7;
+
+export function parseDateTime(string, format, _default) {
+  const century = Math.floor((new Date()).getFullYear() / 100);
+
+  if (string === '') {
+    return null;
+  }
+
+  let parts = format.split(formatSpecifierRegex);
+  let stringIdx = 0;
+  let formatIdx = 0;
+  const result = new Date(0);
+
+  for (let part of parts) {
+    if (!string.startsWith(part, stringIdx)) {
+      return _default;
+    }
+
+    stringIdx += part.length;
+    if (stringIdx === string.length) {
+      if (formatIdx !== format.length) {
+        return _default;
+      }
+      break;
+    }
+
+    formatIdx += part.length + 1;
+
+    if (format[formatIdx] === '%') {
+      if (string[stringIdx++] !== '%') {
+        return _default;
+      }
+    } else {
+      let numberLength = 0;
+      while (digitRegex.test(string[stringIdx + numberLength])) {
+        numberLength++;
+      }
```

Comment:
Weren't you the one screaming how regex are terribly inefficient?

Reply:
Well, yes, but you were using them in a loop. At least this way it is only executed once

Reply:
Idk, I'd still opt for the more explicit version (just because it's already there)

Reply:
It's fine this time since there are not that many numbers in the string so performance will not suffer that much, but I dislike the concept of "just because it's already there", especially when it is not that much code